cocotb/decorators.py

```diff
@@ -76,18 +76,18 @@ def __init__(self, text=""):
         Exception.__init__(self, text)
 
 
-class RunningCoroutine(object):
-    """Per instance wrapper around an function to turn it into a coroutine.
+class RunningTask(object):
+    """Per instance wrapper around a running generator.
 
     Provides the following:
 
-        coro.join() creates a Trigger that will fire when this coroutine
+        task.join() creates a Trigger that will fire when this coroutine
         completes.
 
-        coro.kill() will destroy a coroutine instance (and cause any Join
+        task.kill() will destroy a coroutine instance (and cause any Join
         triggers to fire.
     """
-    def __init__(self, inst, parent):
+    def __init__(self, inst):
         if hasattr(inst, "__name__"):
             self.__name__ = "%s" % inst.__name__
```

Comment:
This does not need to be conditioned, both `cocotb.decorators.coroutine` and async coroutines have `__name__`s. `__str__` will fail if this isn't set. And I would move this after the type check to get better error messages in error cases.